### PR TITLE
#161615259 Integrate a pre-commit hook

### PR DESCRIPTION
**Pull request by kcharles52**
#### What does this PR do?
- Ensures that all code is correctly formatted before being committed
#### Description of Task to be completed?
- In the terminal navigate to hooks folder ` $cd hooks `
- While in the hooks folder run ` $./installhooks `
#### How should this be manually tested?
- Try to commit a file that is not properly formatted
#### What are the relevant pivotal tracker stories?
#161615259

#### Screenshots
- Abort commit on improper code styling

screen shot 2018-11-02 at 13 02 50

- Abort push if tests fail

screen shot 2018-11-02 at 13 06 11

### CODE REVIEW

File: hooks/pre-commit [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+done
+
+if ! $PASS; then
+  echo "\n${RED}COMMIT FAILED:${NC}\n\tYour commit contains files that should pass ESLint but do not. Please fix the ESLint errors and try again.\n\tTo override this validation, use --no-validate"
+  exit 1
+else
+  echo "\n${GREEN}COMMIT SUCCEEDED${NC}"
+fi
+
+exit $?
```
Comment (LehruAngela) [inline hooks/pre-commit]: Add a new line at end of file

File: hooks/readMe.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# How to install hooks
+- After cloning the repository
+- Navigate to the hooks folder : ` cd /path/to/hooks/folder `
+- Run `./installhooks `
```
Comment (LehruAngela) [inline hooks/readMe.md]: Good stuff, however there's a better way to do this. See https://www.npmjs.com/package/husky

**Review by LehruAngela — CHANGES_REQUESTED** [review]

File: hooks/pre-commit (hunk 1 above)
Comment (kcharles52) [reply hooks/pre-commit]: I have removed this file

File: hooks/readMe.md (hunk 2 above)
Comment (kcharles52) [reply hooks/readMe.md]: It was indeed a better way. I have worked with it to do extra hook pre-push

**Comment by LehruAngela** [comment]
You can update the demonstration in the pull request as well

**Comment by kcharles52** [comment]
> You can update the demonstration in the pull request as well

I have updated it @LehruAngela

**Pull request merged**